iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -18,20 +18,24 @@ internal class AmqpUnit : IDisposable
         // If the first argument is set to true, we are disconnecting gracefully via CloseAsync.
         private readonly DeviceIdentity _deviceIdentity;
 
-        private readonly Func<MethodRequestInternal, Task> _methodHandler;
+        private readonly Func<MethodRequestInternal, Task> _onMethodCallback;
         private readonly Action<Twin, string, TwinCollection> _twinMessageListener;
-        private readonly Func<string, Message, Task> _eventListener;
+        private readonly Func<string, Message, Task> _onModuleMessageReceivedCallback;
+        private readonly Func<Message, Task> _onDeviceMessageReceivedCallback;
         private readonly IAmqpConnectionHolder _amqpConnectionHolder;
         private readonly Action _onUnitDisconnected;
         private volatile bool _disposed;
         private volatile bool _closed;
 
-        private readonly SemaphoreSlim _sessionLock = new SemaphoreSlim(1, 1);
+        private readonly SemaphoreSlim _sessionSemaphore = new SemaphoreSlim(1, 1);
 
         private AmqpIoTSendingLink _messageSendingLink;
         private AmqpIoTReceivingLink _messageReceivingLink;
         private readonly SemaphoreSlim _messageReceivingLinkSemaphore = new SemaphoreSlim(1, 1);
 
+        private readonly SemaphoreSlim _messageReceivingCallbackSemaphore = new SemaphoreSlim(1, 1);
+        private bool _deviceReceiveMessageCallbackSet = false;
```

Comment:
rename to isDeviceReceiveMessageCallbackSet

Reply:
sure!